verifiers/internal/gha/provenance.go

```diff
@@ -150,27 +151,32 @@ func verifySha256Digest(prov *intoto.ProvenanceStatement, expectedHash string) e
 
 // VerifyProvenanceSignature returns the verified DSSE envelope containing the provenance
 // and the signing certificate given the provenance and artifact hash.
-func VerifyProvenanceSignature(ctx context.Context, rClient *client.Rekor, provenance []byte, artifactHash string) (*dsselib.Envelope, *x509.Certificate, error) {
-	// Get Rekor entries corresponding to provenance
-	env, cert, err := GetRekorEntriesWithCert(rClient, provenance)
-	if err == nil {
-		return env, cert, nil
+func VerifyProvenanceSignature(ctx context.Context, rClient *client.Rekor,
+	provenance []byte, artifactHash string) (
+	*dsselib.Envelope, *x509.Certificate, error) {
```

Comment:
I notice that you (almost?) always pass around the first two of these return values together, so would it make sense to wrap them in a struct that maps to a higher-level construct?

The triple return value looks odd to me and makes me wonder if the API has room for improvement. (This reaction could be a bias coming from C++, so feel free to ignore!)

Reply:
Nope: I honestly thing we should do that! I think structs are better, and this triple return is pretty annoying.

Reply:
Done! I think that worked out nice: I can re-use it for verifying the signature using the cert and timestamp as well.